packages/api-v2/src/auth/auth.controller.ts

```diff
@@ -85,6 +90,46 @@ export class AuthController {
     return student;
   }
 
+  @Post("forgot-password")
+  public async forgotPassword(
+    @Body() forgotPasswordData: ForgotPasswordDto
+  ): Promise<void> {
+    const student = await this.authService.forgotPassword(forgotPasswordData.email);
+
+    if (student instanceof NoSuchEmail) {
+      throw new BadRequestException(emailDoesNotExistError)
+    }
+    if (student instanceof EmailNotConfirmed) {
+      throw new BadRequestException(emailHasNotBeenConfirmed)
+    }
+  }
+
+  @Post("reset-password")
+  public async resetPassword(
+    @Body() resetPasswordData: ResetPasswordDto
+  ): Promise<void | Error> {
+    const email = await this.authService.decodeResetPassToken(resetPasswordData.token)
+    // Unsure what errors to write here
+    if (email instanceof InvalidPayload) {
+      throw new BadRequestException()
+    } 
+    if (email instanceof BadToken) {
+      throw new BadRequestException()
+    }
+    if (email instanceof TokenExpiredError) {
+      throw new BadRequestException()
+    }
+    if (email instanceof Error) {
+      throw new BadRequestException()
+    }
```

Comment:
these are fine for now, I think TokenExpiredError should return a message so that we can tell the user that.